plugins/techdocs-backend/src/plugin.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DockerContainerRunner,
+  loggerToWinstonLogger,
+  cacheToPluginCacheManager,
+} from '@backstage/backend-common';
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+
+import {
+  Preparers,
+  Generators,
+  Publisher,
+} from '@backstage/plugin-techdocs-node';
+import Docker from 'dockerode';
+import { createRouter } from './service';
+
+/**
+ * The TechDocs plugin is responsible for serving and building documentation for any entity.
+ * @public
+ */
+export const techdocsPlugin = createBackendPlugin({
+  pluginId: 'techdocs-backend',
```

Comment:
```suggestion
  pluginId: 'techdocs',
```

I think techdocs is enough here :)

Reply:
Having this as an alpha export will require some tweaks to package.json as well. #15816 have some context or any existing package with alpha exports.

Reply:
@jhaals Should it be `techdocs` and not `techdocs-backend` if the package name is `techdocs-backend`? 
The `todos-backend` plugin has `pluginId: 'todos-backend'` in its [plugin definition](https://github.com/backstage/backstage/blob/master/plugins/todo-backend/src/plugin.ts#L30), when there are the plugins `todos` and `todos-backend`.

Otherwise, should we also change the pluginId from `todos-backend` to `todos` in this PR for consistency?

Reply:
yeah the todo plugin is the only one that has that suffix - that's a mistake, feel free to change it in this pr or a different one, it doesn't hurt either way